inc/class-wp-cli-command.php

```diff
@@ -59,94 +58,74 @@ public function assets( $args, $args_assoc ) {
 	/**
 	 * Save the contents to disk (or the destination directory).
 	 *
-	 * @synopsis --config=<config> [<path>] [--page-url=<url>] [--replace-from=<from>] [--replace-to=<to>] [--verbose]
+	 * @synopsis [--config=<config>] [<path>] [--page-url=<url>] [--replace-from=<from>] [--replace-to=<to>] [--verbose]
 	 */
 	public function save( $args, $args_assoc ) {
 		$urls = ! empty( $args_assoc['page-url'] ) ? [ $args_assoc['page-url'] ] : get_site_urls( $args_assoc['config'] );
+		$urls = array_unique( $urls );
 
-		$progress = WP_CLI\Utils\make_progress_bar( 'Fetching pages', count( $urls ) );
-		$contents = array_map( function( $url ) use ( $progress, $args_assoc ) {
-			$contents = get_url_contents( $url, $args_assoc['config'] );
-			if ( is_wp_error( $contents ) ) {
-				WP_CLI::warning( $contents->get_error_message() );
-				$contents = '';
-			}
+		$progress = WP_CLI\Utils\make_progress_bar( 'Fetching & Saving pages', count( $urls ) );
 
+		$config = $args_assoc['config'] ?? null;
+		$replace_from = $args_assoc['replace-from'] ?? '';
+		$replace_to = $args_assoc['replace-to'] ?? '';
+		$output_dir = $args[0] ?? '';
+
+		$status = null;
+		foreach( $urls as $url ) {
 			$progress->tick();
-			return $contents;
-		}, $urls );
-		$progress->finish();
 
-		if ( ! empty( $args_assoc['replace-from'] ) ) {
-			add_filter( 'static_page_replace_urls_in_content', function( $contents ) use ( $args_assoc ) {
-				return str_replace( $args_assoc['replace-from'], $args_assoc['replace-to'], $contents );
-			});
-		}
+			$pid = \pcntl_fork();
 
-		foreach ( $contents as &$content ) {
-			$content = replace_urls( $content, $args_assoc['config'] );
-		}
+			if ( $pid == - 1 ) {
+				die( 'could not fork' );
+			} else if ( $pid ) {
+				// This is the main process. Put it to wait till all the child process finished.
+				\pcntl_wait( $status ); //Protect against Zombie children
```

Comment:
Is there a way to design this loop such that we could spawn multiple children at once? It might be that we can do 4 on the CPU ok, so doing them in parallel could give a good perf speedup.